### PR TITLE
Docs: updated Docs for ZIndex Classes

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Docs: updated Docs for ZIndex Classes

### CODE REVIEW

File: docs/src/ZIndex Classes.doc.js [hunk 1]
```diff
@@ -1,181 +1,660 @@
...
-    <Flex alignItems="start" direction="column" gap={4}>
-      <Text>
-        FixedZIndex is used for setting fixed z-index values. FixedZIndex must be instantiated with
-        a number.
-      </Text>
-      <Box padding={2} color="lightGray" rounding={2}>
-        <Markdown
-          text="
+  <MainSection name="FixedZIndex">
+    <MainSection.Subsection columns={2}>
```
Comment (dangerismycat) [inline docs/src/ZIndex Classes.doc.js:22]: Can/should these be full-width?

File: docs/src/ZIndex Classes.doc.js [hunk 2]
```diff
@@ -1,181 +1,660 @@
...
+        description={`
+Extract z-index values from both zIndex classes to use in non-Gestaltifiable HTML Tags. See [z-index in non-Gestalt components](#z-index-in-non-Gestalt-components) to learn more.
 ~~~jsx
-const customButton = <CustomButton zIndex={fixedZindex.index()}/>;
+import { FixedZIndex } from 'gestalt';
+
+const fixedZindex = new FixedZIndex(1);
+
+const canvas = (
+  <canvas
```
Comment (dangerismycat) [inline docs/src/ZIndex Classes.doc.js:217]: This is a great example, btw! 👍

File: docs/src/ZIndex Classes.doc.js [hunk 3]
```diff
@@ -1,181 +1,660 @@
...
-            text="
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection columns={2}>
+      <MainSection.Card
+        cardSize="lg"
+        type="do"
+        description={`
+Use Layer instead of zIndex when a component needs to visually break out of its parent container, for example, in the case of [modals](/Modal), [sheets](/Sheet) and [tooltips](/Tooltip). See [z-Index in foundational components](#zIndex-in-foundational-components) and [ZIndex in Layer](#zIndex-in-Layer) to learn more.
```
Comment (dangerismycat) [inline docs/src/ZIndex Classes.doc.js:233]: I think the Don't example here would be something like
```js
<Box zIndex={1000000}>
  <Modal />
<Box>
```

File: docs/src/ZIndex Classes.doc.js [hunk 4]
```diff
@@ -1,181 +1,660 @@
...
+card(
+  <MainSection name="Variants">
+    <MainSection.Subsection
+      title="zIndex in foundational components"
+      description={`
+[Box](/Box), [Sticky](/Sticky), and [Layer](/Layer) are foundational components that have zIndex props. If any other components need to be positioned within their stacking context, wrap with one of these foundational components to set the z-index.
+
+In the case of Layer, this foundational components creates a new stacking context. Unless there's a conflict with another z-index, don't pass unnecessary zIndex to Layer.
+
+The following example sets a z-index in the Layer wrapping [Sheet](/Sheet) to position Sheet over the page header in the Docs. Set \`HeaderPageZIndex\` below 10 to see the importance of z-index in this example.`}
```
Comment (dangerismycat) [inline docs/src/ZIndex Classes.doc.js:258]: `PAGE_HEADER_ZINDEX`

**Review by dangerismycat — APPROVED** [review]
This looks great! Such a huge improvement over the previous z-index docs. Thanks for this work!

Go ahead and add that Don't example if you agree, then merge this. There are some grammar nits but it'll be faster if I follow-up with a separate PR for those.

File: docs/src/ZIndex Classes.doc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/src/ZIndex Classes.doc.js:22]: YEP! good catch

DONE

File: docs/src/ZIndex Classes.doc.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/src/ZIndex Classes.doc.js:258]: DONE

File: docs/src/ZIndex Classes.doc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/src/ZIndex Classes.doc.js]: we have a max-width on markdown, so this is expected

File: docs/src/ZIndex Classes.doc.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/src/ZIndex Classes.doc.js:233]: DONE

**Pull request merged**